src/uu/top/src/header.rs

```diff
@@ -0,0 +1,283 @@
+    let hardware_interrupt = load[5].parse::<f64>().unwrap_or_default(); // since 2.6.0
+    let software_interrupt = load[6].parse::<f64>().unwrap_or_default(); // since 2.6.0
+    let steal_time = load[7].parse::<f64>().unwrap_or_default(); // since 2.6.11
+                                                                 // GNU do not show guest and guest_nice
+    let guest = load[8].parse::<f64>().unwrap_or_default(); // since 2.6.24
+    let guest_nice = load[9].parse::<f64>().unwrap_or_default(); // since 2.6.33
+    let total = user
+        + nice
+        + system
+        + idle
+        + io_wait
+        + hardware_interrupt
+        + software_interrupt
+        + steal_time
+        + guest
+        + guest_nice;
+
+    format!(
+        "%Cpu(s):  {:.1} us, {:.1} sy, {:.1} ni, {:.1} id, {:.1} wa, {:.1} hi, {:.1} si, {:.1} st",
+        user / total * 100.0,
+        system / total * 100.0,
+        nice / total * 100.0,
+        idle / total * 100.0,
+        io_wait / total * 100.0,
+        hardware_interrupt / total * 100.0,
+        software_interrupt / total * 100.0,
+        steal_time / total * 100.0,
+    )
+}
+
+#[cfg(target_os = "windows")]
+fn cpu() -> String {
+    use libc::malloc;
+    use windows_sys::Wdk::System::SystemInformation::NtQuerySystemInformation;
+
+    #[repr(C)]
+    #[derive(Debug)]
+    struct SystemProcessorPerformanceInformation {
+        idle_time: i64,       // LARGE_INTEGER
+        kernel_time: i64,     // LARGE_INTEGER
+        user_time: i64,       // LARGE_INTEGER
+        dpc_time: i64,        // LARGE_INTEGER
+        interrupt_time: i64,  // LARGE_INTEGER
+        interrupt_count: u32, // ULONG
+    }
+
+    let n_cpu = sysinfo().read().unwrap().cpus().len();
+    let mut cpu_load = SystemProcessorPerformanceInformation {
+        idle_time: 0,
+        kernel_time: 0,
+        user_time: 0,
+        dpc_time: 0,
+        interrupt_time: 0,
+        interrupt_count: 0,
+    };
+    unsafe {
+        let len = n_cpu * size_of::<SystemProcessorPerformanceInformation>();
+        let data = malloc(len);
+        let _ = NtQuerySystemInformation(
+            windows_sys::Wdk::System::SystemInformation::SystemProcessorPerformanceInformation,
+            data,
+            (n_cpu * size_of::<SystemProcessorPerformanceInformation>()) as u32,
+            std::ptr::null_mut(),
+        );
+        for i in 0..n_cpu {
+            let cpu = data.add(i * size_of::<SystemProcessorPerformanceInformation>())
+                as *const SystemProcessorPerformanceInformation;
+            let cpu = cpu.as_ref().unwrap();
+            cpu_load.idle_time += cpu.idle_time;
+            cpu_load.kernel_time += cpu.kernel_time;
+            cpu_load.user_time += cpu.user_time;
+            cpu_load.dpc_time += cpu.dpc_time;
+            cpu_load.interrupt_time += cpu.interrupt_time;
+            cpu_load.interrupt_count += cpu.interrupt_count;
+        }
+    }
```

Comment:
Could you add `// SAFETY` to explain why you use `unsafe` here and why it's safety here?

https://github.com/uutils/coreutils/blob/main/CONTRIBUTING.md#unsafe